### PR TITLE
fix(ACL): correctly remove invalid ids from filter

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to correctly remove invalid ids from filter_var_array
This PR also refresh old legacy codebase

**Fixes** # MON-130416

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 22.10.x
- [x] 23.04.x
- [x] 23.10.x
- [x] 24.04.x
- [x] master

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/include/options/accessLists/reloadACL/reloadACL.php [hunk 1]
```diff
@@ -42,69 +42,85 @@
...
-    $msg->setTimeOut("3");
-    passthru(_CENTREON_PATH_ . "/cron/centAcl.php");
+    $sid = session_id();
+    try {
+        $statement = $pearDB->prepareQuery(
+            <<<SQL
+                UPDATE session SET update_acl = '1' WHERE session_id = :sessionId
+                SQL
+        );
+        $pearDB->executePreparedQuery($statement, ['sessionId' => $sid]);
```
Comment (callapa) [inline centreon/www/include/options/accessLists/reloadACL/reloadACL.php:53]: You should no longer use the methods of the centeronDB class, but those of the extended DBO class.

File: centreon/www/include/options/accessLists/reloadACL/reloadACL.php [hunk 2]
```diff
@@ -42,69 +42,85 @@
...
+        );
+        $msg->info(_("ACL reloaded"));
+        passthru(_CENTREON_PATH_ . "/cron/centAcl.php");
+    } catch (CentreonDbException|\PDOException $e) {
+        $msg->error(_("An error occurred"));
+    }
 } elseif (isset($_POST["o"]) && $_POST["o"] === 'u') {
-    $sel = filter_var_array(
+    $msg = new CentreonMsg();
+    $userIds = filter_var_array(
```
Comment (callapa) [inline centreon/www/include/options/accessLists/reloadACL/reloadACL.php:66]: You can add array_filter directly.
```suggestion
    $userIds = array_filter(filter_var_array(
```

File: centreon/www/include/options/accessLists/reloadACL/reloadACL.php (hunk 1 above)
Comment (callapa) [reply centreon/www/include/options/accessLists/reloadACL/reloadACL.php:53]: Use the PDO methods directly.
CentreonDB method will be deprecated for new dev.

**Review by callapa — APPROVED** [review]

File: centreon/www/include/options/accessLists/reloadACL/reloadACL.php (hunk 2 above)
Comment (jeremyjaouen) [reply centreon/www/include/options/accessLists/reloadACL/reloadACL.php:66]: I prefer not to as filter_var_array can also return null or false

Comment (jeremyjaouen) [reply centreon/www/include/options/accessLists/reloadACL/reloadACL.php:66]: but indeed i didnt check l80 that userIds is actually an array 👀

**Review by wtermellil — APPROVED** [review]

File: centreon/www/include/options/accessLists/reloadACL/reloadACL.php (hunk 2 above)
Comment (callapa) [reply centreon/www/include/options/accessLists/reloadACL/reloadACL.php:66]: You can do this:
```suggestion
    $userIds = array_filter(filter_var_array(...) ?: []);
```

**Pull request merged**